domain/src/test/java/org/oppia/domain/classify/rules/numberinput/NumericInputIsWithinToleranceRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,456 @@
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveIntAnswer_positiveIntInput_negativeIntTolerance_answerNotWithinTolerance() {
+    val inputs = mapOf(
+      "x" to POSITIVE_INT_VALUE_3,
+      "tol" to NEGATIVE_INT_VALUE_1
+    )
+    val matches =
+      inputIsWithinToleranceRuleClassifier
+        .matches(answer = POSITIVE_INT_VALUE_2, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testPositiveIntAnswer_positiveIntInput_zeroTolerance_answerWithinTolerance() {
+    val inputs = mapOf(
+      "x" to POSITIVE_INT_VALUE_1,
+      "tol" to ZERO_REAL_VALUE
+    )
+    val matches =
+      inputIsWithinToleranceRuleClassifier
+        .matches(answer = POSITIVE_INT_VALUE_1, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testPositiveIntAnswer_positiveIntInput_zeroTolerance_answerNotWithinTolerance() {
+    val inputs = mapOf(
+      "x" to POSITIVE_INT_VALUE_3,
+      "tol" to ZERO_INT_VALUE
+    )
+    val matches =
+      inputIsWithinToleranceRuleClassifier
+        .matches(answer = POSITIVE_INT_VALUE_2, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testNegativeIntAnswer_negativeIntInput_zeroTolerance_answerWithinTolerance() {
+    val inputs = mapOf(
+      "x" to NEGATIVE_INT_VALUE_3,
+      "tol" to ZERO_REAL_VALUE
+    )
+    val matches =
+      inputIsWithinToleranceRuleClassifier
+        .matches(answer = NEGATIVE_INT_VALUE_3, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testNegativeIntAnswer_negativeIntInput_zeroTolerance_answerNotWithinTolerance() {
+    val inputs = mapOf(
+      "x" to NEGATIVE_INT_VALUE_3,
+      "tol" to ZERO_REAL_VALUE
+    )
+    val matches =
+      inputIsWithinToleranceRuleClassifier
+        .matches(answer = NEGATIVE_INT_VALUE_2, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testRealAnswer_firstMissingInput_throwsException() {
+    val inputs = mapOf(
+      "tol" to POSITIVE_REAL_VALUE_3_5
+    )
+    val exception = assertThrows(IllegalStateException::class) {
+      inputIsWithinToleranceRuleClassifier
+        .matches(answer = POSITIVE_REAL_VALUE_1_5, inputs = inputs)
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Expected classifier inputs to contain parameter with name 'x' but had: [tol]")
+  }
+
+  @Test
+  fun testRealAnswer_secondMissingInput_throwsException() {
+    val inputs = mapOf(
+      "x" to NEGATIVE_REAL_VALUE_3_5
+    )
+    val exception = assertThrows(IllegalStateException::class) {
+      inputIsWithinToleranceRuleClassifier
+        .matches(answer = POSITIVE_REAL_VALUE_1_5, inputs = inputs)
+    }
+
+    assertThat(exception)
+      .hasMessageThat()
+      .contains("Expected classifier inputs to contain parameter with name 'tol' but had: [x]")
+  }
+
+  @Test
+  fun testRealAnswer_firstStringInput_throwsException() {
```

Comment:
Add a test case with 3 inputs as well and I guess i wasn't clear enough in my previous comments but what i wanted was to have kept the existing test cases in which there was incorrect input variable like `c` or anything

Reply:
If I test with 3 inputs, exception is not getting caught

Reply:
@aggarwalpulkit596 Also isn't that case covered in missing inputs and incorrect inputs ?

Reply:
@prayutsu what's the error for 3 inputs?

Reply:
![Screenshot from 2020-09-30 15-40-07](https://user-images.githubusercontent.com/54636525/94672693-3efe4a00-0333-11eb-9a48-0bb5bb972a9e.png)

Reply:
I think this will not work because, currently exception message is of the form ```Expected classifier inputs to contain parameter with name 'x' but had: [c, tol]``` but in case of three inputs it can't display the range

Reply:
@MohamedMedhat1998 What do you think?

Reply:
@prayutsu having test cases that check the case where we have 3 inputs is a good idea.  
I suggest trying to pass a `map` that contains both "x" and "tol" + some other value ("y" for example) and observe which exception you will get. After you know the exception, you can write a test case that checks whether this exception is thrown or not.  
It's kinda weird to gerenrate an exception then test whether it is generated or not :sweat_smile:, but I think it is ok.

Reply:
@MohamedMedhat1998 I tried same, but see the exception in the screenshot, it does not throw the exception of the "same type" that we get in other test cases

Reply:
> @aggarwalpulkit596 I think this will not work because, currently exception message is of the form `Expected classifier inputs to contain parameter with name 'x' but had: [c, tol]` but in case of three inputs it can't display the range

Please, see this comment @MohamedMedhat1998

Reply:
> @MohamedMedhat1998 I tried the same, but see the exception in the screenshot, it does not throw the exception of the "same type" that we get in other test cases

Can you share your code, please?